javascript/app-framework/src/components/DataShaperApp/DataShaperApp.hooks.ts

```diff
@@ -51,20 +67,87 @@ export function useOnChangeWidth(
 	).run
 }
 
-export function useHandlerArgs(selectedKey: string | undefined): {
-	handler?: string | undefined
-	args?: string[] | undefined
-} {
-	return useMemo(() => {
-		if (selectedKey != null) {
-			const tokens = selectedKey?.split('|')
-			if (tokens.length > 0) {
-				return {
-					handler: tokens[0],
-					args: tokens.slice(1),
+export interface GeneratedRoute {
+	path: string
+	renderer: React.ComponentType
+	props: any
+}
+
+export function useDataPackageResourceRoutes(
+	plugins: Map<string, DataShaperAppPlugin>,
+): GeneratedRoute[] {
+	const dp = useDataPackage()
+	const [result, setResult] = useState<GeneratedRoute[]>([])
+	useEffect(() => {
+		setResult(getRoutes(dp, plugins))
+		return dp.onChange(() => setResult(getRoutes(dp, plugins)))
+	}, [dp, plugins])
+	return result
+}
+
+function getRoutes(
+	dp: DataPackage,
+	plugins: Map<string, DataShaperAppPlugin>,
+): GeneratedRoute[] {
+	const result: GeneratedRoute[] = []
+	addSources(result, dp.resources, plugins)
+	return result
+}
+
+function addSources(
+	result: GeneratedRoute[],
+	resources: SchemaResource[],
+	plugins: Map<string, DataShaperAppPlugin>,
+	root = '/resource',
+) {
+	for (const resource of resources) {
+		// TODO: check plugins
+		const renderer =
+			DEFAULT_HANDLERS[resource.profile] ||
+			plugins.get(resource.profile)?.renderer
+
+		if (renderer != null) {
+			result.push({
+				path: `${root}/${resource.name}`,
+				renderer: renderer as any,
+				props: { resource },
+			})
+
+			// Special case for raw resources embedded in datatable under `path` property
+			// and or datatable.json
+			if (isDataTableResource(resource)) {
+				if (resource.path != null && typeof resource.path === 'string') {
+					const pathItems = resource.path.split('/') ?? []
+					const lastPathItem = pathItems[pathItems.length - 1]
+					result.push({
+						path: `${root}/${resource.name}/${lastPathItem}`,
```

Comment:
I am a little torn on what the path should end up looking like. One consideration is that if the path contains things that are renamable by the user, bookmarks can eventually break. So we could use a guid that is generated for each resource. It also seems like we might want to consider adding some intermediary strings to indicate categories where relevant, like `${root}/resources/${resource.id}`?